mythril/analysis/module/base.py

```diff
@@ -0,0 +1,94 @@
+""" Mythril Detection Modules
+
+This module includes an definition of the DetectionModule interface.
+DetectionModules implement different analysis rules to find weaknesses and vulnerabilities.
+"""
+import logging
+from typing import List, Set, Optional
+
+from mythril.analysis.report import Issue
+from mythril.laser.ethereum.state.global_state import GlobalState
+
+from abc import ABC, abstractmethod
+from enum import Enum
+
+# Get logger instance
+log = logging.getLogger(__name__)
+
+
+class EntryPoint(Enum):
+    """ EntryPoint Enum
+
+    This enum is used to signify the entry_point of detection modules.
+    See also the class documentation of DetectionModule
+    """
+
+    POST = 1
+    CALLBACK = 2
+
+
+class DetectionModule(ABC):
+    """The base detection module.
+
+    All custom-built detection modules must inherit from this class.
+
+    There are several class properties that expose information about the detection modules
+    - name: The name of the detection module
+    - swc_id: The SWC ID associated with the weakness that the module detects
+    - description: A description of the detection module, and what it detects
+    - entry_point: Mythril can run callback style detection modules, or modules that search the statespace.
+                [IMPORTANT] POST entry points severely slow down the analysis, try to always use callback style modules
+    - pre_hooks: A list of instructions to hook the laser vm for (pre execution of the instruction)
+    - post_hooks: A list of instructions to hook the laser vm for (post execution of the instruction)
+    """
+
+    name = "Detection Module Name"
+    swc_id = "SWC-000"
+    description = "Detection module description"
+    entry_point = EntryPoint.POST  # type: EntryPoint
```

Comment:
it makes sense if it's defaulted to CALLBACK right? as all the modules now are callback.